test/Window/WindowBase.test.cpp

```diff
@@ -219,5 +221,30 @@ TEST_CASE("[Window] sf::WindowBase", runDisplayTests())
 
         // Should compile if user provides both a specific handler and a catch-all
         windowBase.handleEvents([](const sf::Event::Closed&) {}, [](const auto&) {});
+
+        // Should compile if user provides a handler taking an event subtype by value or reference,
+        // but not rvalue reference because it would never be called.
+        windowBase.handleEvents([](sf::Event::Closed) {});
+        windowBase.handleEvents([](const sf::Event::Closed) {});
+        windowBase.handleEvents([](sf::Event::Closed&) {});
+        windowBase.handleEvents([](const sf::Event::Closed&) {});
+
+        // Should compile if user provides a handler with deleted rvalue ref-qualified call operator
+        struct LvalueOnlyHandler
+        {
+            void operator()(const sf::Event::Closed&) &
+            {
+            }
+            void operator()(const sf::Event::Closed&) && = delete;
+        };
+        windowBase.handleEvents(LvalueOnlyHandler{});
+
+        // Should compile if user provides a reference to a move-only handler
+        auto handler = [p = std::make_unique<int>()](const sf::Event::Closed&) {};
+        windowBase.handleEvents(handler);
+        windowBase.handleEvents(std::as_const(handler));
```

Comment:
This is test and corresponding implementation is actually not about move-only, but just about passing by reference. I will simplify.

Reply:
Done.